.Jenkins/workflows/Jenkinsfile

```diff
@@ -54,6 +54,47 @@ pipeline {
                         }
                     }
                 }
+                stage('pylint') {
+                    agent {
+                        node {
+                            label 'docker'
+                            customWorkspace "${WORKSPACE}/${STAGE_NAME}"
+                        }
+                    }
+                    steps {
+                        script {
+                            // DOCKER_IMAGE is defined through Jenkins project
+                            pylintStageDockerImage="${DOCKER_IMAGE}_${BUILD_NUMBER}_${STAGE_NAME}"
+                        }
+                        echo "cleanup workspace"
+                        sh 'for f in $(ls -A); do rm -rf ${f}; done'
+                        // DE_REPO is defined through Jenkins project
+                        echo "clone decisionengine code from ${DE_REPO}"
+                        sh '''
+                            git clone ${DE_REPO}
+                            cd decisionengine
+                            echo "checkout ${BRANCH} branch"
+                            git checkout ${BRANCH}
+                            echo GITHUB_PR_NUMBER: ${GITHUB_PR_NUMBER} - GITHUB_PR_STATE: ${GITHUB_PR_STATE}
+                            if [[ -n ${GITHUB_PR_NUMBER} && ${GITHUB_PR_STATE} == OPEN ]]; then
+                                git fetch origin pull/${GITHUB_PR_NUMBER}/merge:merge${GITHUB_PR_NUMBER}
+                                git checkout merge${GITHUB_PR_NUMBER}
+                            fi
+                            cd ..
+                        '''
+                        echo "prepare docker image ${pylintStageDockerImage}"
+                        sh "docker build --pull --tag ${pylintStageDockerImage} --build-arg BASEIMAGE=hepcloud/decision-engine-ci:${BRANCH} --build-arg UID=\$(id -u) --build-arg GID=\$(id -g) -f decisionengine/package/ci/Dockerfile decisionengine/package/ci/"
+                        echo "Run ${STAGE_NAME} tests"
+                        sh "docker run --rm --env PYTEST_TIMEOUT=${PYTEST_TIMEOUT} -v ${WORKSPACE}/decisionengine:${WORKSPACE}/decisionengine -w ${WORKSPACE}/decisionengine ${pylintStageDockerImage} \"pylint\" \"pylint.log\""
```

Comment:
Because there is other python code outside `src` I thought to let pylint to run on all of them.
But if the code to check is only in `src` folder I'm fine with this change.

Reply:
Using the suggested change, i.e. run `python3 -m pylint src/` works well.
Errors and the exit status are reported properly.
The only issue that shows up by running `pylint` this way is the following error
```
 ************* Module src.decisionengine.framework
src/decisionengine/framework/__init__.py:1:0: E0401: Unable to import 'src.decisionengine.framework.about' (import-error)
```
that was not reported when running pylint on file by file basis.
I'm not sure how to address this issue.
If this is a real error, I'm happy to make the change you are proposing.

Reply:
.... Interesting

I'm a bit puzzled by the `src` at the front of the thiny there.  Does the error drop if we run against `src/decisionengine` rather than just `src/` ?

Reply:
We can ignore the `pyproject.toml` for now.

Reply:
yes, if I run pylint against `src/decisionengine` no errors show up.

Reply:
I'd go with that and see if we can keep everything nice and simple ;)

Reply:
What about running it as
`python3 -m pylint src/decisionengine/ src/tests/`
just to test also the python test file in `src/tests/` ?

Reply:
works for me